app/Http/Controllers/ImportController.php

```diff
@@ -34,4 +35,93 @@ public function store(Request $request) {
 
         return redirect()->route('imports.index');
     }
+
+    public function getPrepare(Import $import) {
+        $this->authorize('modify', $import);
+
+        $headers = [];
+
+        $file = fopen(storage_path('app/imports/' . $import->file), 'r');
+
+        while ($row = fgetcsv($file, 999, ',')) {
+            foreach ($row as $column) {
+                $headers[] = $column;
+            }
+
+            break;
+        }
+
+        return view('imports.prepare', compact('headers'));
+    }
+
+    public function postPrepare(Request $request, Import $import) {
+        $this->authorize('modify', $import);
+
+        $request->validate([
+            'column_happened_on' => 'required|integer',
+            'column_description' => 'required|integer',
+            'column_amount' => 'required|integer',
+        ]);
+
+        $import->fill([
+            'column_happened_on' => $request->input('column_happened_on'),
+            'column_description' => $request->input('column_description'),
+            'column_amount' => $request->input('column_amount'),
+            'status' => 1
+        ])->save();
+
+        return redirect()->route('imports.index');
+    }
+
+    public function getComplete(Import $import) {
+        $this->authorize('modify', $import);
+
+        $file = fopen(storage_path('app/imports/' . $import->file), 'r');
+
+        $heading = true;
+        $rows = [];
+
+        while ($row = fgetcsv($file, 999, ',')) {
+            if (!$heading) {
+                $rows[] = [
+                    'happened_on' => $row[$import->column_happened_on],
+                    'description' => $row[$import->column_description],
+                    'amount' => $row[$import->column_amount]
+                ];
+            }
+
+            $heading = false;
```

Comment:
Why is this not in the if statement? Why do you need to update that it's false every row?